src/ModelMerger.py

```diff
@@ -0,0 +1,158 @@
+                matches_for_met = sorted_matches[sorted_matches['source_namespace'] == met_id]
+                # Assign the best available match
+                for _, row in matches_for_met.iterrows():
+                    match_id = row['target_namespace']
+                    if (met_id not in best_matches.keys()) and (match_id not in best_matches.values()):
+                        best_matches[met_id] = match_id
+                        #print(met_id + ": " + match_id + ", " + str(row[score_type]))
+                #else:
+                #    # If no match is found, keep the original ID
+                #    best_matches[met_id] = met_id
+                #    print(met_id + ": " + met_id)
+
+        # apply the id translation to the metabolite and re-appends the translation compartment suffix
+        for met in to_translate:
+            try:
+                met.id = best_matches[met.id] + "_t"
+            except KeyError:
+                met.id = met.id + "_t"
+
+    def translate_ids(self, score_thr, score_type="Name_score"):
+        """
+        Translates metabolite IDs in self.cobra_model based on a score threshold and updates corresponding
+        translation reaction IDs.
+
+        Parameters:
+            score_thr (float): Minimum score required for ID translation.
+            score_type (str): Score type to filter matches by, must be a valid column in the matches table
+
+        Raises:
+            ValueError: If the specified score type is not found in the matches table or if a metabolite has
+                        more than one or no "TR_" reaction.
+        """
+        if score_type not in self.matches.columns: # makes sure that the chosen score is specified in the matches table
+            raise ValueError("Specified score type: " + score_type + "doesn't exist")
+        cobra_model = self.memo_model.cobra_model
+        reliable_matches = self.matches.loc[self.matches[score_type] >= score_thr]
+        #  selects the metabolites to be translated
+        to_translate = list()
+        for met in cobra_model.metabolites:
+            if met.compartment == "t":
+                to_translate = to_translate + [met]
+        # applies the translation to the metabolites
+        self.translate_metabolites(to_translate, reliable_matches, score_type)
+        #  translate the ids of the respective TR_ reactions
+        for met in cobra_model.metabolites:
+            if met.compartment == "t":
+                tr_rxns_for_current_met = [rxn for rxn in met.reactions if rxn.id.startswith("TR_")]
+                if len(tr_rxns_for_current_met) != 1:
+                    raise ValueError(met.id + " should have one and only one TR_ reaction, it has " +
+                                     str(len(tr_rxns_for_current_met)) + " instead")
+                tr_rxn = tr_rxns_for_current_met[0]
+                tr_rxn.id = "TR_" + met.id
+
+    def create_exchanges(self):
+        cobra_model = self.memo_model.cobra_model
```

Comment:
Add a comment what this is doing and why (because we lost the actual EX_ in the "first" function